src/client/sandbox/node/live-node-list/dom-mutation-tracker.js

```diff
@@ -0,0 +1,77 @@
+import nativeMethods from '../../native-methods';
+import { getTagName, isShadowUIElement } from '../../../utils/dom';
+import getNativeQuerySelectorAll from '../../../utils/get-native-query-selector-all';
+
+const MAX_SAFE_INTEGER = Math.pow(2, 53) - 1;
+const MIN_SAFE_INTEGER = -MAX_SAFE_INTEGER;
+
+class DOMMutationTracker {
+    constructor () {
+        this._mutations          = nativeMethods.objectCreate.call(window.Object, null);
+        this._isDomContentLoaded = false;
+
+        nativeMethods.addEventListener.call(document, 'DOMContentLoaded', () => {
+            this._isDomContentLoaded = true;
+        });
+    }
+
+    _updateVersion (tagName) {
+        if (tagName in this._mutations) {
+            if (this._mutations[tagName] === MAX_SAFE_INTEGER)
+                this._mutations[tagName] = MIN_SAFE_INTEGER;
+            else
+                ++this._mutations[tagName];
+        }
+        else
+            this._mutations[tagName] = MIN_SAFE_INTEGER;
+    }
+
+    _processElement (el) {
+        if (!el.tagName || isShadowUIElement(el))
+            return;
+
+        const tagName = getTagName(el);
+
+        this._updateVersion('*');
+        this._updateVersion(tagName);
+    }
+
+    _processChildren (el) {
+        if (!el.querySelectorAll)
```

Comment:
Add test for this case

Reply:
https://github.com/DevExpress/testcafe-hammerhead/pull/1322/files#diff-2511acd18edd5a40b5ab70fe3ef86068R191